ui/libs/search/src/lib/state/filters/filters.service.ts

```diff
@@ -0,0 +1,134 @@
+import { Inject, Injectable } from '@angular/core';
+import { FiltersRepository } from './filters.repository';
+import { HttpClient } from '@angular/common/http';
+import { CommonSettings, ESS_SETTINGS } from '@eosc-search-service/common';
+import { forkJoin, map, Observable, of, tap } from 'rxjs';
+import { getEntity, updateEntities } from '@ngneat/elf-entities';
+import { assertNotNull, HashMap, IHasId } from '@eosc-search-service/types';
+import {
+  ICollectionSearchMetadata,
+  ISolrQueryParams,
+  toSolrQueryParams,
+} from '../results';
+import { ISet } from '../../sets';
+import {addFacetsToFilter, IFilterEntry, mapFacetToTreeNodes} from './filters.model';
+import {
+  FILTERS_FETCH_MORE_LIMIT,
+  INITIAL_FILTER_OPTION_COUNT,
+  ISearchResults,
+} from '../common';
+import { arrayAdd } from '@ngneat/elf/lib/props-array-factory';
+
+@Injectable({ providedIn: 'root' })
+export class FiltersService {
+  readonly url: string;
+  constructor(
+    protected _repository: FiltersRepository,
+    protected http: HttpClient,
+    @Inject(ESS_SETTINGS) protected settings: CommonSettings
+  ) {
+    this.url = `/${this.settings.backendApiPath}/${this.settings.search.apiPath}`;
+  }
+
+  get$() {}
+
+  loadNext$() {}
+
+  search$(query: string) {}
+
+  showMore$(
+    id: string,
+    queryParams: HashMap<unknown>,
+    activeSet: ISet
+  ): Observable<unknown> {
+    this._repository.store.update(
+      updateEntities(id, (state) => ({ ...state, showMore: !state.showMore }))
+    );
+
+    return this.searchFilters$(id, '*', queryParams, activeSet, true);
+  }
+
+  searchFilters$(
+    id: string,
+    query: string,
+    queryParams: HashMap<unknown>,
+    activeSet: ISet,
+    fetchNext: boolean = false
+  ) {
+    const params = toSolrQueryParams(queryParams);
+
+    const filter = this._repository.store.query(getEntity(id));
+    assertNotNull(filter);
+    if (!fetchNext) {
+      this._repository.store.update(
+        updateEntities(id, (f) => {
+          const cursors = { ...f.cursors };
+          Object.entries(cursors).forEach(([key, val]) => {
+            cursors[key] = { offset: 0, moreAvailable: true };
+          });
+          return {
+            ...f,
+            cursors,
+          };
+        })
+      );
+    }
+    return forkJoin<IFilterEntry[][]>(
+      activeSet.collections
+        .filter((collection) => filter.cursors[collection.type]?.moreAvailable)
+        .map((metadata) => this.fetchMoreForSingle$(id, metadata, params, fetchNext))
+    );
+  }
+
+  fetchMoreForSingle$<T extends IHasId>(
+    id: string,
+    metadata: ICollectionSearchMetadata<T>,
+    params: ISolrQueryParams,
+    fetchNext: boolean = false
+  ): Observable<IFilterEntry[]> {
+    const _params = { ...params, ...metadata.params, rows: 0 };
```

Comment:
Could you add a comment on why have you used `rows = 0` to keep it for the future?